tests/meltano/core/block/test_extract_load.py

```diff
@@ -621,19 +622,51 @@ async def test_elb_with_job_context(
 
 
 class TestExtractLoadUtils:
-    def test_generate_job_name(self):
-        """Verify that the job id is generated correctly when an environment is provided."""
+    def test_generate_state_id(self):
+        """Verify that a state ID is generated correctly given an active environment and optional suffix."""
         block1 = mock.Mock(spec=IOBlock)
         block1.string_id = "block1"
 
         block2 = mock.Mock(spec=IOBlock)
         block2.string_id = "block2"
 
         project = mock.Mock()
+        project.active_environment = Environment(name="test")
+
+        assert (
+            generate_state_id(project, "suffix", block1, block2)
+            == "test:block1-to-block2:suffix"
+        )
+
+        assert (
+            generate_state_id(project, "suffix", block1, block2)
+            == "test:block1-to-block2:suffix"
+        )
```

Comment:
```suggestion
        )
        
        assert (
            generate_state_id(project, None, block1, block2)
            == "test:block1-to-block2"
        )

```

Worth testing no suffix just incase ?

Reply:
Oops, that was my original intention... :sweat_smile: https://github.com/meltano/meltano/pull/6758/commits/fc2af322ff9d9aa1ef37b906af609b265ebf8b3d